airflow/serialization/serialized_objects.py

```diff
@@ -784,7 +724,7 @@ def default_serialization(cls, strict, var) -> str:
         return str(var)
 
     @classmethod
-    def deserialize(cls, encoded_var: Any, use_pydantic_models=False) -> Any:
+    def deserialize(cls, encoded_var: Any) -> Any:
```

Comment:
The `use_pydantic_models` used in python operator https://github.com/apache/airflow/blob/main/providers/src/airflow/providers/standard/utils/python_virtualenv_script.jinja2#L81, might be changes required in here aswell?

Reply:
Yes. That's why I think **some** TODOs will stay after this one as we are currently using Context Pydantic version to pass to PythonOperator - this one will have to be replaced with whatever Context serialization TaskSDK has (cc: @ashb @kaxil -> I have not looked at this yet, but since we are using HTTPs to send the context to SDK and Pydantic classes for Task SDK under-the hood, so I guess we will be able to re-use what's there. 

For now I think I will just have to copy privately the Pydantic classes needed to the Python Operator, to make it possible to use standard operator, but we will eventually have to do it "better" - lilkely after this PR is merged.

Reply:
One of the reasons I was not in a super-hurry with this one.

Reply:
We haven't tackled context yet, but these models can be removed and we'll add back the ones we need in the execution SDK (some will be very different, some similar) so merge when ever you are happy with this